### PR TITLE
linux: fix building with clang

### PR DESCRIPTION
**Pull request by wsnipex**
## Description

building with clang on linux is broken since https://github.com/xbmc/xbmc/commit/6589fcd97d90fcdd3e907550f9d2e36b8d454dd4

also fixes a warning in bufferobject, I'm not 100% sure about, see commit description

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by howie-f — APPROVED** [review]
fixes build and the warning flags with clang version 11.0.1-2

it looks like `-Wno-deprecated-copy` is also a candidate https://github.com/xbmc/xbmc/pull/20719#issuecomment-1010250007

**Comment by wsnipex** [comment]
> fixes build and the warning flags with clang version 11.0.1-2
> 
> it looks like `-Wno-deprecated-copy` is also a candidate [#20719 (comment)](https://github.com/xbmc/xbmc/pull/20719#issuecomment-1010250007)

added

**Comment by thexai** [comment]
Jenkins log on Android ARM64 confirms that is fixed:

```
22:11:34 -- The CXX compiler identification is Clang 9.0.9
22:11:34 -- The C compiler identification is Clang 9.0.9
```

Now is clean of warnings again:

```
23:02:40 [ 58%] Built target dvdvideocodecs
23:02:41 [ 58%] Building CXX object build/cores/VideoPlayer/CMakeFiles/VideoPlayer.dir/VideoPlayerVideo.cpp.o
23:02:50 [ 58%] Building CXX object build/cores/VideoPlayer/demuxers/CMakeFiles/dvddemuxers.dir/DVDDemuxFFmpeg.cpp.o
23:02:57 [ 58%] Building CXX object build/utils/CMakeFiles/utils.dir/GLUtils.cpp.o
23:03:00 [ 58%] Building CXX object build/cores/VideoPlayer/inputstreams/CMakeFiles/dvdinputstreams.dir/DVDFactoryInputStream.cpp.o
23:03:11 [ 59%] Building CXX object build/cores/VideoPlayer/CMakeFiles/VideoPlayer.dir/VideoReferenceClock.cpp.o
23:03:15 [ 59%] Building CXX object build/cores/VideoPlayer/inputstreams/CMakeFiles/dvdinputstreams.dir/DVDInputStream.cpp.o
23:03:17 [ 59%] Linking CXX static library utils.a
23:03:18 [ 59%] Built target utils
23:03:18 [ 59%] Building CXX object build/cores/VideoPlayer/subtitles/CMakeFiles/dvdsubtitles.dir/DVDFactorySubtitle.cpp.o
23:03:21 [ 59%] Building CXX object build/cores/VideoPlayer/inputstreams/CMakeFiles/dvdinputstreams.dir/DVDInputStreamFFmpeg.cpp.o
23:03:22 [ 59%] Building CXX object build/cores/VideoPlayer/demuxers/CMakeFiles/dvddemuxers.dir/DVDDemuxUtils.cpp.o
23:03:23 [ 59%] Building CXX object build/cores/VideoPlayer/subtitles/CMakeFiles/dvdsubtitles.dir/DVDSubtitleLineCollection.cpp.o
23:03:25 [ 59%] Building CXX object build/cores/VideoPlayer/subtitles/CMakeFiles/dvdsubtitles.dir/DVDSubtitleParserMicroDVD.cpp.o
23:03:27 [ 59%] Building CXX object build/cores/VideoPlayer/inputstreams/CMakeFiles/dvdinputstreams.dir/DVDInputStreamFile.cpp.o
23:03:31 [ 59%] Linking CXX static library VideoPlayer.a
23:03:32 [ 59%] Built target VideoPlayer
23:03:32 [ 59%] Building CXX object build/cores/VideoPlayer/subtitles/CMakeFiles/dvdsubtitles.dir/DVDSubtitleParserMPL2.cpp.o
23:03:35 [ 59%] Building CXX object build/cores/VideoPlayer/inputstreams/CMakeFiles/dvdinputstreams.dir/DVDInputStreamMemory.cpp.o
23:03:40 [ 59%] Building CXX object build/cores/VideoPlayer/inputstreams/CMakeFiles/dvdinputstreams.dir/DVDInputStreamNavigator.cpp.o
23:03:41 [ 59%] Building CXX object build/cores/VideoPlayer/demuxers/CMakeFiles/dvddemuxers.dir/DVDDemuxVobsub.cpp.o
23:03:44 [ 59%] Building CXX object build/cores/VideoPlayer/subtitles/CMakeFiles/dvdsubtitles.dir/DVDSubtitleParserSami.cpp.o
23:03:46 [ 59%] Building CXX object build/cores/VideoPlayer/subtitles/CMakeFiles/dvdsubtitles.dir/DVDSubtitleParserSubrip.cpp.o
23:03:50 [ 59%] Building CXX object build/cores/VideoPlayer/subtitles/CMakeFiles/dvdsubtitles.dir/DVDSubtitleParserVplayer.cpp.o
23:03:50 [ 59%] Building CXX object build/cores/VideoPlayer/demuxers/CMakeFiles/dvddemuxers.dir/DVDFactoryDemuxer.cpp.o
23:03:51 [ 59%] Building CXX object build/cores/VideoPlayer/subtitles/CMakeFiles/dvdsubtitles.dir/DVDSubtitleStream.cpp.o
23:03:54 [ 59%] Building CXX object build/cores/VideoPlayer/subtitles/CMakeFiles/dvdsubtitles.dir/DVDSubtitlesLibass.cpp.o
23:04:07 [ 60%] Building CXX object build/cores/VideoPlayer/inputstreams/CMakeFiles/dvdinputstreams.dir/DVDInputStreamStack.cpp.o
23:04:11 [ 60%] Linking CXX static library dvddemuxers.a
```

**Review by thexai — APPROVED** [review]

File: xbmc/utils/BufferObject.h [hunk 1]
```diff
@@ -28,6 +28,7 @@ class CBufferObject : public IBufferObject
    */
   static std::unique_ptr<CBufferObject> GetBufferObject(bool needsCreateBySize);
 
+  virtual bool CreateBufferObject(uint32_t format, uint32_t width, uint32_t height) override = 0;
```
Comment (lrusak) [inline xbmc/utils/BufferObject.h:31]: Should probably just be
```suggestion
  bool CreateBufferObject(uint32_t format, uint32_t width, uint32_t height) override { return false; }
```

**Pull request merged**